### PR TITLE
Jellyfish: init at 2.3.0

### PR DESCRIPTION
**Pull request by giang-nghg**
###### Motivation for this change

Add Jellyfish, a common k-mer counter in bioinformatics.

###### Things done

- Built on platform(s)
  - [x] x86_64-linux
  - [ ] aarch64-linux
  - [ ] x86_64-darwin
  - [ ] aarch64-darwin
- [ ] For non-Linux: Is `sandbox = true` set in `nix.conf`? (See [Nix manual](https://nixos.org/manual/nix/stable/#sec-conf-file))
- [ ] Tested via one or more NixOS test(s) if existing and applicable for the change (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
- [x] Tested compilation of all packages that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review wip"`
- [x] Tested execution of all binary files (usually in `./result/bin/`)
- [21.11 Release Notes (or backporting 21.05 Release notes)](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md#generating-2111-release-notes)
  - [ ] (Package updates) Added a release notes entry if the change is major or breaking
  - [ ] (Module updates) Added a release notes entry if the change is significant
  - [ ] (Module addition) Added a release notes entry if adding a new NixOS module
- [x] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md).

### CODE REVIEW

File: maintainers/maintainer-list.nix [hunk 1]
```diff
@@ -13000,5 +13000,11 @@
     github = "mbprtpmnr";
     githubId = 88109321;
   };
+  giang = {
```
Comment (SuperSandro2000) [inline maintainers/maintainer-list.nix:13003]: Please put this into a separate commit with the message ``maintainers: add NAME`` and sort it alphabetically.

File: pkgs/top-level/all-packages.nix [hunk 2]
```diff
@@ -33497,4 +33497,6 @@ with pkgs;
   zthrottle = callPackage ../tools/misc/zthrottle { };
 
   zktree = callPackage ../applications/misc/zktree {};
+
+  jellyfish = callPackage ../applications/science/biology/jellyfish {};
```
Comment (SuperSandro2000) [inline pkgs/top-level/all-packages.nix:33501]: Please sort this entry.

File: pkgs/applications/science/biology/jellyfish/default.nix [hunk 3]
```diff
@@ -0,0 +1,31 @@
...
+
+stdenv.mkDerivation rec {
+    pname = "jellyfish";
+    version = "2.3.0";
+
+    src = fetchurl {
+        url = "https://github.com/gmarcais/Jellyfish/releases/download/v${version}/${pname}-${version}.tar.gz";
+        sha256 = "e195b7cf7ba42a90e5e112c0ed27894cd7ac864476dc5fb45ab169f5b930ea5a";
+    };
+
```
Comment (SuperSandro2000) [inline pkgs/applications/science/biology/jellyfish/default.nix:11]: Do we need to do something else? Seems to be a little to little to do.

**Review by SuperSandro2000 — COMMENTED** [review]
Please format the package file with nixpkgs.

File: pkgs/applications/science/biology/jellyfish/default.nix (hunk 3 above)
Comment (giang-nghg) [reply pkgs/applications/science/biology/jellyfish/default.nix:11]: @SuperSandro2000 It seems Nix automatically runs `configure` and `make`? This is all I need to do on Linux to get it built and symlinked :thinking:

File: maintainers/maintainer-list.nix (hunk 1 above)
Comment (giang-nghg) [reply maintainers/maintainer-list.nix:13003]: @SuperSandro2000 I've already put it in a separated commit, or do you mean another PR as well?

Comment (giang-nghg) [reply maintainers/maintainer-list.nix:13003]: @SuperSandro2000 https://github.com/NixOS/nixpkgs/pull/146663

File: pkgs/applications/science/biology/jellyfish/default.nix [hunk 4]
```diff
@@ -0,0 +1,31 @@
+{ stdenv, fetchurl, lib }:
+
+stdenv.mkDerivation rec {
+    pname = "jellyfish";
+    version = "2.3.0";
+
+    src = fetchurl {
+        url = "https://github.com/gmarcais/Jellyfish/releases/download/v${version}/${pname}-${version}.tar.gz";
```
Comment (Synthetica9) [inline pkgs/applications/science/biology/jellyfish/default.nix:8]: Why not use fetchFromGitHub here?

Comment (giang-nghg) [reply pkgs/applications/science/biology/jellyfish/default.nix:8]: @Synthetica9 I tried but it didn't work (I forgot what the error was, can rerun it again and post it here if needed)

Comment (Synthetica9) [reply pkgs/applications/science/biology/jellyfish/default.nix:8]: Please do.

Comment (giang-nghg) [reply pkgs/applications/science/biology/jellyfish/default.nix:8]: @Synthetica9 oh I remembered. I always get hash mismatch. How do you get the right hash? I did:
```
find path/to/folder -type f -print0 | sort -z | xargs -0 sha256sum | sha256sum
```

Comment (Synthetica9) [reply pkgs/applications/science/biology/jellyfish/default.nix:8]: I just insert a wrong SHA256 hash (all 0's) and copy the correct hash from the error message, it's called TOFU (trust on first use)

Comment (giang-nghg) [reply pkgs/applications/science/biology/jellyfish/default.nix:8]: @Synthetica9 ah I tried that before, but I copied the `wanted` instead of the `got` :facepalm: . Thanks!

**Comment by giang-nghg** [comment]
waiting for https://github.com/NixOS/nixpkgs/pull/146717

File: pkgs/applications/science/biology/jellyfish/default.nix [hunk 5]
```diff
@@ -21,9 +29,9 @@ stdenv.mkDerivation rec {
             Guillaume Marcais and Carl Kingsford, A fast, lock-free approach for efficient parallel counting of occurrences of k-mers. Bioinformatics (2011) 27(6): 764-770 (first published online January 7, 2011) doi:10.1093/bioinformatics/btr011
         '';
         branch = "master";
-        homepage = "https://github.com/gmarcais/Jellyfish";
-        downloadPage = "https://github.com/gmarcais/Jellyfish/releases";
-        changelog = "https://github.com/gmarcais/Jellyfish/blob/master/CHANGES";
+        homepage = "https://github.com/${owner}/${repo}";
+        downloadPage = "https://github.com/${owner}/${repo}/releases";
+        changelog = "https://github.com/${owner}/${repo}/blob/master/CHANGES";
```
Comment (SuperSandro2000) [inline pkgs/applications/science/biology/jellyfish/default.nix:34]: Please revert this. Clickable links make researching things way easier.

File: pkgs/applications/science/biology/jellyfish/default.nix [hunk 6]
```diff
@@ -1,14 +1,22 @@
...
+        repo = pname;
+        rev = "v${version}";
+        sha256 = "1d9f0ni2ll1basc5p0shwb4gq1iz1ahnzv2imfgb93bihyvcmq66";
     };
 
+    buildInputs = [ autoconf automake libtool gettext pkg-config ];
+
+    preConfigure = ''
+        autoreconf -i
+    '';
```
Comment (SuperSandro2000) [inline pkgs/applications/science/biology/jellyfish/default.nix:18]:
```suggestion
    nativeBuildInputs = [ autoreconfHook gettext pkg-config ];
```
is gettext used as a binary or cli tool? If it is the first it needs to go to buildInputs.

**Comment by thyol** [comment]
@giang-nghg Do you plan to finish this? If not, I would take over.

**Comment by giang-nghg** [comment]
@thyol yes, please do. Thank you :)

**Comment by Lord-Valen** [comment]
@tyhol @gian-nghg Do either of you plan on continuing this?

**Comment by SuperSandro2000** [comment]
closing because of staleness and the author indicated that it should be taken over
